### PR TITLE
Generate Views for geometric types in `ArborX`

### PR DESCRIPTION
**Pull request by JBludau**
This pr adds bindings for 1D `Kokkos::View` types of the `ArborX` geometry classes. Having them as part of the static instead of the dynamic part is easier and can be compiled once for the bindings.
 
The exposed bindings do only cover the minimal required set of functionalities.
Ideally `pykokkos-base` would allow other libraries to pass some special types and would generate bindings for them. Otherwise downstream libs like us always have to reinvent the wheel. 

The `create_mirror_view` and 'deep_copy' methods are member functions of the respective `View` at the moment as pybind11 has problems with the overloading resolution for these functions. I will update with the solution in `pykokkos-base`.

based on #1 and #2 main changes are in `static/CMakeLists.txt` and the `.in` files

**Does not compile as I apparently mixed up some namespace stuff**

### CODE REVIEW

**Review by aprokop — APPROVED** [review]
Compiles.

File: static/CMakeLists.txt [hunk 1]
```diff
@@ -6,16 +6,43 @@ set(pyArborX_STATIC_UTILITY_MODULES Experimental)
...
 # get source files for selected static modules
 foreach(PY_MODULE IN LISTS pyArborX_STATIC_GEOMETRY_MODULES pyArborX_STATIC_UTILITY_MODULES)
-  list(APPEND PY_SRC ${PROJECT_SOURCE_DIR}/src/pyArborX_${PY_MODULE}.cpp)
+  list(APPEND pyArborX_SRC ${PROJECT_SOURCE_DIR}/src/pyArborX_${PY_MODULE}.cpp)
   list(APPEND MODULE_INCLUDES "\n#include \"pyArborX_${PY_MODULE}.hpp\"" )
   list(APPEND MODULE_GENERATORS "\npyArborX::generate${PY_MODULE}Wrapper(m)")
 endforeach()
 
+#generate bindings for views of ArborX types
+set(LAYOUTS LayoutLeft LayoutRight)
```
Comment (dalg24) [inline static/CMakeLists.txt:15]: Why would we care to do both layout?

Comment (dalg24) [reply static/CMakeLists.txt:15]: These are rank-1 views

Comment (JBludau) [reply static/CMakeLists.txt:15]: reason was that I do not know when we will have an option in pykokkos-base to generate view-bindings for the ArborX types. If this extension to pykokkos-base takes longer we still can easily generate other view variants for our types.

**Pull request merged**